<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Overview of RCy3 • RCy3</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Overview of RCy3">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RCy3</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Cancer-networks-and-data.html">Cancer networks and data</a>
    </li>
    <li>
      <a href="../articles/Custom-Graphics.html">Custom Graphics and Labels</a>
    </li>
    <li>
      <a href="../articles/Cytoscape-and-NDEx.html">Cytoscape and NDEx</a>
    </li>
    <li>
      <a href="../articles/Cytoscape-and-graphNEL.html">Cytoscape and graphNEL</a>
    </li>
    <li>
      <a href="../articles/Cytoscape-and-iGraph.html">Cytoscape and igraph</a>
    </li>
    <li>
      <a href="../articles/Filtering-Networks.html">Filtering Networks</a>
    </li>
    <li>
      <a href="../articles/Group-nodes.html">Group nodes</a>
    </li>
    <li>
      <a href="../articles/Identifier-mapping.html">Identifier mapping</a>
    </li>
    <li>
      <a href="../articles/Importing-data.html">Importing data</a>
    </li>
    <li>
      <a href="../articles/Network-functions-and-visualization.html">Network functions and visualization</a>
    </li>
    <li>
      <a href="../articles/Overview-of-RCy3.html">Overview of RCy3</a>
    </li>
    <li>
      <a href="../articles/Phylogenetic-trees.html">Phylogenetic-trees</a>
    </li>
    <li>
      <a href="../articles/Upgrading-existing-scripts.html">Upgrading existing scripts</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cytoscape/RCy3">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Overview of RCy3</h1>
                        <h4 class="author">by Alexander Pico, Tanja Muetze, Georgi Kolishovski, Paul Shannon</h4>
            
            <h4 class="date">2020-02-29</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cytoscape/RCy3/blob/master/vignettes/Overview-of-RCy3.Rmd"><code>vignettes/Overview-of-RCy3.Rmd</code></a></small>
      <div class="hidden name"><code>Overview-of-RCy3.Rmd</code></div>

    </div>

    
    
<p><em>Cytoscape</em> is a well-known bioinformatics tool for displaying and exploring biological networks. <strong>R</strong> is a powerful programming language and environment for statistical and exploratory data analysis. <em>RCy3</em> uses CyREST to communicate between <strong>R</strong> and Cytoscape, allowing any graphs (e.g., igraph, graphNEL or dataframes) to be viewed, explored and manipulated with the Cytoscape point-and-click visual interface. Thus, via RCy3, these two quite different, quite useful bioinformatics software environments are connected, mutually enhancing each other, providing new possibilities for exploring biological data.</p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="cf">if</span>(<span class="op">!</span><span class="st">"RCy3"</span> <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span>()){</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">    <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">    BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"RCy3"</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">}</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(RCy3)</a></code></pre></div>
</div>
<div id="prerequisites" class="section level1">
<h1 class="hasAnchor">
<a href="#prerequisites" class="anchor"></a>Prerequisites</h1>
<p>In addition to this package (RCy3), you will need:</p>
<ul>
<li>
<strong>Cytoscape 3.6.1</strong> or greater, which can be downloaded from <a href="http://www.cytoscape.org/download.php" class="uri">http://www.cytoscape.org/download.php</a>. Simply follow the installation instructions on screen.</li>
</ul>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>First, launch Cytoscape and keep it running whenever using RCy3. Confirm that you have everything installed and running:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">    <span class="kw"><a href="../reference/cytoscapePing.html">cytoscapePing</a></span> ()</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">    <span class="kw"><a href="../reference/cytoscapeVersionInfo.html">cytoscapeVersionInfo</a></span> ()</a></code></pre></div>
</div>
<div id="my-first-network" class="section level1">
<h1 class="hasAnchor">
<a href="#my-first-network" class="anchor"></a>My first network</h1>
<p>Let’s create a Cytoscape network from some basic R objects</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">nodes &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="dt">id=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"node 0"</span>,<span class="st">"node 1"</span>,<span class="st">"node 2"</span>,<span class="st">"node 3"</span>),</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">           <span class="dt">group=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A"</span>,<span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"B"</span>), <span class="co"># categorical strings</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">           <span class="dt">score=</span><span class="kw"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">15</span>,<span class="dv">5</span>)), <span class="co"># integers</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">           <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">edges &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="dt">source=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"node 0"</span>,<span class="st">"node 0"</span>,<span class="st">"node 0"</span>,<span class="st">"node 2"</span>),</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">           <span class="dt">target=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"node 1"</span>,<span class="st">"node 2"</span>,<span class="st">"node 3"</span>,<span class="st">"node 3"</span>),</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">           <span class="dt">interaction=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"inhibits"</span>,<span class="st">"interacts"</span>,<span class="st">"activates"</span>,<span class="st">"interacts"</span>),  <span class="co"># optional</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">           <span class="dt">weight=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5.1</span>,<span class="fl">3.0</span>,<span class="fl">5.2</span>,<span class="fl">9.9</span>), <span class="co"># numeric</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9">           <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="kw"><a href="../reference/createNetworkFromDataFrames.html">createNetworkFromDataFrames</a></span>(nodes,edges, <span class="dt">title=</span><span class="st">"my first network"</span>, <span class="dt">collection=</span><span class="st">"DataFrame Example"</span>)</a></code></pre></div>
<div id="switch-styles" class="section level2">
<h2 class="hasAnchor">
<a href="#switch-styles" class="anchor"></a>Switch styles</h2>
<p>Check out the marquee style!</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/setVisualStyle.html">setVisualStyle</a></span>(<span class="st">'Marquee'</span>)</a></code></pre></div>
</div>
<div id="my-own-style" class="section level2">
<h2 class="hasAnchor">
<a href="#my-own-style" class="anchor"></a>My own style</h2>
<p>Create your own style with node attribute fill mappings and some defaults</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">style.name =<span class="st"> "myStyle"</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">defaults &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">NODE_SHAPE=</span><span class="st">"diamond"</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                 <span class="dt">NODE_SIZE=</span><span class="dv">30</span>,</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                 <span class="dt">EDGE_TRANSPARENCY=</span><span class="dv">120</span>,</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                 <span class="dt">NODE_LABEL_POSITION=</span><span class="st">"W,E,c,0.00,0.00"</span>)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">nodeLabels &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mapVisualProperty.html">mapVisualProperty</a></span>(<span class="st">'node label'</span>,<span class="st">'id'</span>,<span class="st">'p'</span>)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">nodeFills &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mapVisualProperty.html">mapVisualProperty</a></span>(<span class="st">'node fill color'</span>,<span class="st">'group'</span>,<span class="st">'d'</span>,<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A"</span>,<span class="st">"B"</span>), <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#FF9900"</span>,<span class="st">"#66AAAA"</span>))</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">arrowShapes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mapVisualProperty.html">mapVisualProperty</a></span>(<span class="st">'Edge Target Arrow Shape'</span>,<span class="st">'interaction'</span>,<span class="st">'d'</span>,<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"activates"</span>,<span class="st">"inhibits"</span>,<span class="st">"interacts"</span>),<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Arrow"</span>,<span class="st">"T"</span>,<span class="st">"None"</span>))</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">edgeWidth &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mapVisualProperty.html">mapVisualProperty</a></span>(<span class="st">'edge width'</span>,<span class="st">'weight'</span>,<span class="st">'p'</span>)</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="kw"><a href="../reference/createVisualStyle.html">createVisualStyle</a></span>(style.name, defaults, <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(nodeLabels,nodeFills,arrowShapes,edgeWidth))</a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="kw"><a href="../reference/setVisualStyle.html">setVisualStyle</a></span>(style.name)</a></code></pre></div>
<p><em>Pro-tip: if you want to set NODE_WIDTH and NODE_HEIGHT independently, you also need to unlock the node dimensions with…</em></p>
<pre><code><a href="../reference/lockNodeDimensions.html">lockNodeDimensions(FALSE, style.name)</a></code></pre>
</div>
</div>
<div id="bioconductor-graph-example" class="section level1">
<h1 class="hasAnchor">
<a href="#bioconductor-graph-example" class="anchor"></a>Bioconductor graph example</h1>
<p>Alternatively, you might want to start from a Bioconductor graphNEL object. Here we create a 4-node graph in R, send it to Cytoscape for display and layout. For the sake of simplicity, no node attributes and no visual styles are included; those topics are covered in subsequent steps.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">    g =<span class="st"> </span><span class="kw">new</span> (<span class="st">'graphNEL'</span>, <span class="dt">edgemode=</span><span class="st">'directed'</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/graph/man/addNode.html">addNode</a></span> (<span class="st">'A'</span>, g)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/graph/man/addNode.html">addNode</a></span> (<span class="st">'D'</span>, g)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/graph/man/addNode.html">addNode</a></span> (<span class="st">'C'</span>, g, <span class="dt">edges =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">'D'</span>))</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">    g =<span class="st"> </span>graph<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/graph/man/addNode.html">addNode</a></span> (<span class="st">'B'</span>, g, <span class="dt">edges =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'A'</span>,<span class="st">'D'</span>,<span class="st">'C'</span>)))</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">    <span class="kw"><a href="../reference/createNetworkFromGraph.html">createNetworkFromGraph</a></span> (g, <span class="dt">title=</span><span class="st">'simple network'</span>, <span class="dt">collection=</span><span class="st">'GraphNEL Example'</span>)</a></code></pre></div>
<p>You should now have the structure of this 4-node graph with a basic, default style. Fortunately, Cytoscape has some built-in rendering rules in which (and unless instructed otherwise) nodes and edges are rendered and a default (user-preference) layout algorithm is applied.</p>
<div id="add-node-attributes" class="section level2">
<h2 class="hasAnchor">
<a href="#add-node-attributes" class="anchor"></a>Add node attributes</h2>
<p>We often know quite a lot about the nodes and edges in our graphs. By conveying this information visually, the graph will be easier to explore. For instance, we may know that protein A phosphorylates protein B, that A is a kinase and B a transcription factor, and that their mRNA expression (compared to a control) is a log2 fold change of 1.8 and 3.2 respectively. One of the core features of Cytoscape is visual styles, which allow you to specify how data values (e.g., <code>kinase',</code>transcription factor’; expression ratios) should be conveyed in the visual properties of the graph (e.g., node shape, node color or size).</p>
<p>We continue with the simple 4-node graph, adding two kinds data values (<code>moleculeType' and</code>log2fc’). The easiest way to do this is via data.frames. However, you can also include attributes together with the original graph models as Bioconductor graphs, igraphs or data.frames and then use the provided <em>create</em> functions to create and load in a single step (see <em>createNetworkFromGraph</em>, <em>createNetworkFromIgraph</em> and <em>createNetworkFromDataFrames</em> functions). Check out the other vignettes for more exampls.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">    df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span> (<span class="dt">moleculeType=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'kinase'</span>,<span class="st">'TF'</span>,<span class="st">'cytokine'</span>,<span class="st">'cytokine'</span>),</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">                     <span class="dt">log2fc=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1.8</span>,<span class="fl">3.0</span>,<span class="op">-</span><span class="fl">1.2</span>,<span class="op">-</span><span class="fl">2.5</span>),</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                     <span class="dt">row.names =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'A'</span>,<span class="st">'B'</span>,<span class="st">'C'</span>,<span class="st">'D'</span>), <span class="co"># row.names = node names</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                     <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)       <span class="co"># important when loading strings!</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5">    <span class="kw"><a href="../reference/loadTableData.html">loadTableData</a></span> (df)</a></code></pre></div>
<p>Note that adding the attributes does not in itself cause the appearance of the graph to change. Such a change requires that you specify and apply visual style mappings, which will be explained in the <em>next</em> section. You can, however, examine these attributes in Cytoscape, using Cytoscape’s the <strong>Data Panel</strong> to display data values associated with selected nodes immediately below the Cytoscape window.</p>
</div>
<div id="modifying-the-display-defaults-and-mappings" class="section level2">
<h2 class="hasAnchor">
<a href="#modifying-the-display-defaults-and-mappings" class="anchor"></a>Modifying the display: defaults and mappings</h2>
<p>RCy3 provides an easy way to not only change the default styles, but more interestingly, RCy3 also provides easy access to <em>mapping</em> your data to visual styles, e.g., allowing the size, shape and color of nodes and edges to be determined by the data you have associated with those nodes and edges.</p>
<p>First, let’s change the the defaults.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">    <span class="kw"><a href="../reference/setNodeShapeDefault.html">setNodeShapeDefault</a></span> (<span class="st">'OCTAGON'</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">    <span class="kw"><a href="../reference/setNodeColorDefault.html">setNodeColorDefault</a></span> (<span class="st">'#AAFF88'</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">    <span class="kw"><a href="../reference/setNodeSizeDefault.html">setNodeSizeDefault</a></span>  (<span class="dv">60</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">    <span class="kw"><a href="../reference/setNodeFontSizeDefault.html">setNodeFontSizeDefault</a></span> (<span class="dv">30</span>)</a></code></pre></div>
<p>Now we will add some visual mappings. Let’s map `moleculeType’ to node shapes. First, we can see which shapes are available in Cytoscape, then we can define the mapping with paired lists.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">    <span class="kw"><a href="../reference/getNodeShapes.html">getNodeShapes</a></span> ()   <span class="co"># diamond, ellipse, trapezoid, triangle, etc.</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">    column &lt;-<span class="st"> 'moleculeType'</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">    values &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="st">'kinase'</span>,  <span class="st">'TF'</span>,<span class="st">'cytokine'</span>)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">    shapes &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="st">'DIAMOND'</span>, <span class="st">'TRIANGLE'</span>, <span class="st">'RECTANGLE'</span>)</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">    <span class="kw"><a href="../reference/setNodeShapeMapping.html">setNodeShapeMapping</a></span> (column, values, shapes)</a></code></pre></div>
<p>The node shape mapping is an example of a <em>discrete</em> mapping, where a style is defined for each, discrete value. This is useful for categorical data (like type) where there is only a limited set of possible values. This is in contast to the other two other types of mappings: <em>continuous</em> and <em>passthrough</em>. In the case of expression values, for example, we will want to use <em>continuous</em> mapping (e.g., to node color), defining a small set of control points, rather than an explicit color for each possible data value. Cytoscape will simply interpolate between the control points to provide a gradient of colors. Let’s try that one now</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">    column &lt;-<span class="st"> 'log2fc'</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">    control.points &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="op">-</span><span class="fl">3.0</span>, <span class="fl">0.0</span>, <span class="fl">3.0</span>)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">    colors &lt;-<span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="st">'#5588DD'</span>, <span class="st">'#FFFFFF'</span>, <span class="st">'#DD8855'</span>)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">    <span class="kw"><a href="../reference/setNodeColorMapping.html">setNodeColorMapping</a></span> (column, control.points, colors)</a></code></pre></div>
<p>Note that there are three colors and three control points. However, you can also specify <em>two additional</em> colors beyond the number of control points if you want to set extreme (or out-of-bounds) colors for values less than or greater than your control points.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">    control.points &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="op">-</span><span class="fl">2.0</span>, <span class="fl">0.0</span>, <span class="fl">2.0</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">    colors &lt;-<span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="st">'#2255CC'</span>, <span class="st">'#5588DD'</span>, <span class="st">'#FFFFFF'</span>, <span class="st">'#DD8855'</span>,<span class="st">'#CC5522'</span>)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">    <span class="kw"><a href="../reference/setNodeColorMapping.html">setNodeColorMapping</a></span> (column, control.points, colors)</a></code></pre></div>
<p>Now, add a node size rule, using <em>log2fc</em> again as controlling node values.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">    control.points =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="op">-</span><span class="fl">3.0</span>, <span class="fl">2.0</span>, <span class="fl">3.0</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">    sizes     =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span> (<span class="dv">20</span>, <span class="dv">80</span>, <span class="dv">90</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">    <span class="kw"><a href="../reference/setNodeSizeMapping.html">setNodeSizeMapping</a></span> (column, control.points, sizes)</a></code></pre></div>
<p>If you recall the third type of mapping, <em>passthrough</em>, we can see it already working in our current network example. The node labels! By default, the <em>name</em> column is mapped to the node label property using <em>passthrough</em> logic: the value is passed directly to the style property.</p>
</div>
<div id="selecting-nodes" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-nodes" class="anchor"></a>Selecting nodes</h2>
<p>Let us now try selecting nodes in Cytoscape from R. Select the C node by name:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">    <span class="kw"><a href="../reference/selectNodes.html">selectNodes</a></span> (<span class="st">'C'</span>,<span class="st">'name'</span>)</a></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">    <span class="kw"><a href="../reference/getSelectedNodes.html">getSelectedNodes</a></span> ()</a></code></pre></div>
<p>Now we wish to extend the selected nodes to include the first neighbors of the already-selected node B. This is a common operation: for instance, after selecting one or more nodes based on experimental data or annotation, you may want to explore these in the context of interaction partners (in a protein-protein network) or in relation to upstream and downstream partners in a signaling or metabolic network. Type:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">    <span class="kw"><a href="../reference/selectFirstNeighbors.html">selectFirstNeighbors</a></span> ()</a></code></pre></div>
<p>You will see that three nodes are now selected. Get their names back to R as a list:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">    node.names &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getSelectedNodes.html">getSelectedNodes</a></span> ()</a></code></pre></div>
<p>And, finally, deselection works as you’d expect by means of a general <em>clearSelection</em> function:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">    <span class="kw"><a href="../reference/clearSelection.html">clearSelection</a></span>()</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">    ?clearSelection</a></code></pre></div>
</div>
<div id="saving-and-export" class="section level2">
<h2 class="hasAnchor">
<a href="#saving-and-export" class="anchor"></a>Saving and export</h2>
<p>Session files save <em>everything</em>. As with most project software, we recommend saving often!</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">    <span class="kw"><a href="../reference/saveSession.html">saveSession</a></span>(<span class="st">'vignette_session'</span>) <span class="co">#.cys</span></a></code></pre></div>
<p><strong>Note</strong>: If you don’t specify a complete path, the files will be saved relative to your current working directory in R.</p>
<div id="saving-high-resolution-image-files" class="section level3">
<h3 class="hasAnchor">
<a href="#saving-high-resolution-image-files" class="anchor"></a>Saving high resolution image files</h3>
<p>You can export extremely high resolution images, including vector graphic formats.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">    full.path=<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>(),<span class="st">'vignette_image'</span>,<span class="dt">sep=</span><span class="st">'/'</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">    <span class="kw"><a href="../reference/exportImage.html">exportImage</a></span>(full.path, <span class="st">'PNG'</span>, <span class="dt">zoom=</span><span class="dv">200</span>) <span class="co">#.png scaled by 200%</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3">    <span class="kw"><a href="../reference/exportImage.html">exportImage</a></span>(full.path, <span class="st">'PDF'</span>) <span class="co">#.pdf</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4">    ?exportImage</a></code></pre></div>
</div>
</div>
</div>
<div id="browse-available-functions-commands-and-arguments" class="section level1">
<h1 class="hasAnchor">
<a href="#browse-available-functions-commands-and-arguments" class="anchor"></a>Browse available functions, commands and arguments</h1>
<p>RCy3 functions</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/help.html">help</a></span>(<span class="dt">package=</span>RCy3)</a></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="84%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th>Category</th>
<th>Description</th>
<th>Examples</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Apps</strong></td>
<td>Inspecting and managing apps for Cytoscape.</td>
<td><em>installApp() disableApp() getInstalledApps()</em></td>
</tr>
<tr class="even">
<td><strong>Collections</strong></td>
<td>Getting information about network collections.</td>
<td><em>getCollectionList() getCollectionNetworks()</em></td>
</tr>
<tr class="odd">
<td><strong>Commands</strong></td>
<td>Constructing any arbitrary CyREST API or Commands API method via standard GET, PUT, POST and DELETE protocols.</td>
<td><em>cyrestGET() commandsPOST() cyrestAPI() commandsRun()</em></td>
</tr>
<tr class="even">
<td><strong>CyNDEx</strong></td>
<td>Communicating with NDEx from within Cytoscape.</td>
<td><em>importNetworkFromNDEx() exportNetworkToNDEx()</em></td>
</tr>
<tr class="odd">
<td><strong>Cytoscape System</strong></td>
<td>Checking Cytoscape System information, including versions and memory usage.</td>
<td><em>cytoscapePing() cytoscapeVersionInfo()</em></td>
</tr>
<tr class="even">
<td><strong>Filters</strong></td>
<td>Selecting nodes and edges based on filter criteria.</td>
<td><em>createDegreeFilter() createColumnFilter()</em></td>
</tr>
<tr class="odd">
<td><strong>Groups</strong></td>
<td>Working with groups in Cytoscape.</td>
<td><em>createGroup() collapseGroup()</em></td>
</tr>
<tr class="even">
<td><strong>Layouts</strong></td>
<td>Performing layouts in addition to getting and setting layout properties.</td>
<td><em>layoutNetwork() getLayoutNames()</em></td>
</tr>
<tr class="odd">
<td><strong>Networks</strong></td>
<td>Creating and managing networks and retrieving information on networks, nodes and edges.</td>
<td><em>createNetworkFrom…() create…FromNetwork() getNetworkSuid(), exportNetwork() getAllNodes() getEdgeCount(), getFirstNeighbors()</em></td>
</tr>
<tr class="even">
<td><strong>Network Selection</strong></td>
<td>Manipulating selection of nodes and edges in networks.</td>
<td><em>selectNodes() invertNodeSelection() selectFirstNeighbors()</em></td>
</tr>
<tr class="odd">
<td><strong>Network Views</strong></td>
<td>Performing view operations in addition to getting and setting view properties.</td>
<td><em>getCurrentView() fitContent() exportImage() toggleGraphicsDetails()</em></td>
</tr>
<tr class="even">
<td><strong>Session</strong></td>
<td>Managing Cytoscape sessions, including save, open and close.</td>
<td><em>openSession() saveSession() closeSession()</em></td>
</tr>
<tr class="odd">
<td><strong>Style Bypasses</strong></td>
<td>Setting and clearing bypass values for visual properties.</td>
<td><em>setNodeColorBypass() setEdgeLineStyleBypass() hideNodes()</em></td>
</tr>
<tr class="even">
<td><strong>Style Defaults</strong></td>
<td>Getting and setting default values for visual properties.</td>
<td><em>setNodeShapeDefault() setEdgeLineWidthDefault()</em></td>
</tr>
<tr class="odd">
<td><strong>Style Dependencies</strong></td>
<td>Getting and setting style dependencies.</td>
<td><em>lockNodeDimensions()</em></td>
</tr>
<tr class="even">
<td><strong>Style Mappings</strong></td>
<td>Defining mappings between table column values and visual properties.</td>
<td><em>mapVisualProperty() updateStyleMapping() setNodeSizeMapping() setEdgeColorMapping()</em></td>
</tr>
<tr class="odd">
<td><strong>Styles</strong></td>
<td>Managing styles and retrieving general lists of properties relevant to multiple style modes.</td>
<td><em>createVisualStyle() setVisualStyle() exportVisualStyles() getArrowShapes()</em></td>
</tr>
<tr class="even">
<td><strong>Style Values</strong></td>
<td>Retrieving current values for visual properties.</td>
<td><em>getNodeWidth() getEdgeColor() getNetworkZoom()</em></td>
</tr>
<tr class="odd">
<td><strong>Tables</strong></td>
<td>Managing table columns and table column functions, like map and rename, as well as loading and extracting table data in Cytoscape.</td>
<td><em>getTableColumns() renameTableColumn() loadTableData() mapTableColumn()</em></td>
</tr>
<tr class="even">
<td><strong>Tools</strong></td>
<td>Performing actions found in the Tools Menu in Cytoscape.</td>
<td><em>cybrowserDialog() diffusionBasic()</em></td>
</tr>
<tr class="odd">
<td><strong>User Interface</strong></td>
<td>Controling the panels in the Cytoscape user interface.</td>
<td><em>hidePanel() floatPanel() dockPanel()</em></td>
</tr>
</tbody>
</table>
<p>Open swagger docs for live instances of CyREST API and Commands API:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="../reference/cyrestAPI.html">cyrestAPI</a></span>()  <span class="co"># CyREST API</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw"><a href="../reference/commandsAPI.html">commandsAPI</a></span>()  <span class="co"># Commands API</span></a></code></pre></div>
<p>List available commands and arguments in R. Use “help” to list top level:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw"><a href="../reference/commandsHelp.html">commandsHelp</a></span>(<span class="st">"help"</span>)  </a></code></pre></div>
<p>List <strong>network</strong> commands. Note that “help” is optional:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="../reference/commandsHelp.html">commandsHelp</a></span>(<span class="st">"help network"</span>)  </a></code></pre></div>
<p>List arguments for the <strong>network select</strong> command:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw"><a href="../reference/commandsHelp.html">commandsHelp</a></span>(<span class="st">"help network select"</span>) </a></code></pre></div>
<p>That covers the basics of network manipulation. Check out the other vignettes for additional amd more complex examples. And when you are ready to work with some real data, check out the other basic and advanced R tutorials, <a href="https://github.com/cytoscape/cytoscape-automation/tree/master/for-scripters/R" class="uri">https://github.com/cytoscape/cytoscape-automation/tree/master/for-scripters/R</a>.</p>
</div>
<div id="more-examples" class="section level1">
<h1 class="hasAnchor">
<a href="#more-examples" class="anchor"></a>More examples</h1>
<p>Don’t forget to check out the other vignettes in this package:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes</a></span>(<span class="st">"RCy3"</span>)</a></code></pre></div>
<p>In addition, the Cytoscape team is collecting scripts from the community in a public GitHub repository at <a href="https://github.com/cytoscape/cytoscape-automation/tree/master/for-scripters/R" class="uri">https://github.com/cytoscape/cytoscape-automation/tree/master/for-scripters/R</a>.</p>
</div>
<div id="development" class="section level1">
<h1 class="hasAnchor">
<a href="#development" class="anchor"></a>Development</h1>
<p>The RCy3 project code and documentation is maintained at GitHub: <a href="https://github.com/cytoscape/RCy3" class="uri">https://github.com/cytoscape/RCy3</a>. All bugs and feature requests are tracked as <strong>Issues</strong>, <a href="https://github.com/cytoscape/RCy3/issues" class="uri">https://github.com/cytoscape/RCy3/issues</a>.</p>
</div>
<div id="credits" class="section level1">
<h1 class="hasAnchor">
<a href="#credits" class="anchor"></a>Credits</h1>
<ul>
<li>Paul Shannon’s generous advice and mentorship was very important for transforming this package from using XMLRPC and CytoscapeRPC to using CyREST.</li>
<li>David Otasek, Keiichiro Ono and Barry Demchak kindly provided CyREST as well as help and support for new functionalities and changes.</li>
<li>Mark Grimes and Ruth Isserlin kindly provided helpful user feedback.</li>
<li>Julia Gustavsen generously developed various use cases/examples for using RCy3 with biological data during GSOC 2016, <a href="https://github.com/jooolia/gsoc_Rcy3_vignettes/blob/master/blog_post_drafts/final_work_submission.md" class="uri">https://github.com/jooolia/gsoc_Rcy3_vignettes/blob/master/blog_post_drafts/final_work_submission.md</a>.</li>
<li>Tanja Muetze provided many years of development, design, maintenance and documentation for the RCy3 project.</li>
<li>All contributors, new and old, are dynamically acknowledged in our <strong>Contributor Graph</strong>, <a href="https://github.com/cytoscape/RCy3/graphs/contributors" class="uri">https://github.com/cytoscape/RCy3/graphs/contributors</a>
</li>
</ul>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<ol style="list-style-type: decimal">
<li>Shannon P, Markiel A, Ozier O, Baliga NS, Wang JT, Ramage D, Amin N, Schwikowski B, Ideker T. 2003. Cytoscape: a software environment for integrated models of biomolecular interaction networks. Genome Res. Nov;13(11):2498-504</li>
<li>Huber W, Carey VJ, Long L, Falcon S, Gentleman R. 2007. Graphs in molecular biology. BMC Bioinformatics. 2007 Sep 27;8.</li>
<li>Ono K, Muetze T, Kolishovski G, Shannon P, Demchak, B. CyREST: Turbocharging Cytoscape Access for External Tools via a RESTful API [version 1; referees: 2 approved]. F1000Research 2015, 4:478.</li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#prerequisites">Prerequisites</a></li>
      <li><a href="#getting-started">Getting started</a></li>
      <li>
<a href="#my-first-network">My first network</a><ul class="nav nav-pills nav-stacked">
<li><a href="#switch-styles">Switch styles</a></li>
      <li><a href="#my-own-style">My own style</a></li>
      </ul>
</li>
      <li>
<a href="#bioconductor-graph-example">Bioconductor graph example</a><ul class="nav nav-pills nav-stacked">
<li><a href="#add-node-attributes">Add node attributes</a></li>
      <li><a href="#modifying-the-display-defaults-and-mappings">Modifying the display: defaults and mappings</a></li>
      <li><a href="#selecting-nodes">Selecting nodes</a></li>
      <li><a href="#saving-and-export">Saving and export</a></li>
      </ul>
</li>
      <li><a href="#browse-available-functions-commands-and-arguments">Browse available functions, commands and arguments</a></li>
      <li><a href="#more-examples">More examples</a></li>
      <li><a href="#development">Development</a></li>
      <li><a href="#credits">Credits</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alex Pico, Tanja Muetze, Paul Shannon.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
